homeassistant/components/airvisual_pro/__init__.py

```diff
@@ -46,7 +77,21 @@ async def async_get_data() -> dict[str, Any]:
     )
 
     await coordinator.async_config_entry_first_refresh()
-    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = coordinator
+    hass.data.setdefault(DOMAIN, {})[entry.entry_id] = AirVisualProData(
+        coordinator=coordinator, node=node
+    )
+
+    async def async_shutdown(_: Event) -> None:
+        """Define an event handler to disconnect from the websocket."""
+        nonlocal reload_task
+        if reload_task:
+            with suppress(asyncio.CancelledError):
```

Comment:
It probably isn't likely that we'll encounter this during shutdown, but this shouldn't hurt anything.